pkg/image/containerdregistry/resolver.go

```diff
@@ -85,14 +87,31 @@ func credential(cfg *configfile.ConfigFile) func(string) (string, string, error)
 	}
 }
 
-func loadConfig(dir string) (*configfile.ConfigFile, error) {
+func loadConfig(dir string) (cfg *configfile.ConfigFile, err error) {
```

Comment:
Nit: any reason to add the named return values? This opens the door for [naked returns](https://go.dev/tour/basics/7), which generally reduce the readability of a function.